### PR TITLE
Enable cancellation flows for Akismet products

### PR DESCRIPTION
**Pull request by jboland88**
## Proposed Changes

* This PR enables cancellation flows for Akismet products from the purchases page

## Testing Instructions

* You will need a sample Akismet purchase to test this flow
* Apply this PR on your local Calypso testing environment
* Navigate to `/me/purchases` and select your Akismet purchase
* From the manage purchase page, you should be able to use the auto-renewal toggle to turn auto-renew off and on again

Screenshot 2023-03-23 at 4 04 58 PM

* Check that there are no console errors while changing the auto-renew setting
* Now, click on "Cancel Subscription" at the bottom of the management screen

Screenshot 2023-03-23 at 4 05 03 PM

* This will take you to the purchase cancellation page

Screenshot 2023-03-23 at 4 05 10 PM

* Click on "Cancel Purchase"

Screenshot 2023-03-23 at 4 05 18 PM

* You will be presented with a survey screen. Click "Submit and cancel product" to proceed
* The product should be cancelled successfully. If you are still within the refund window (14 days) the subscription will also be removed from your account.

Regression testing
* Test with both Jetpack and WPCOM purchases to ensure the Auto-renew toggle still works as expected. Particularly, for a disconnected Jetpack site, the toggle should not be present.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/components/marketing-survey/cancel-auto-renewal-form/index.jsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ import './style.scss';
 class CancelAutoRenewalForm extends Component {
 	static propTypes = {
 		purchase: PropTypes.object.isRequired,
-		selectedSite: PropTypes.object.isRequired,
+		selectedSiteId: PropTypes.number.isRequired,
```
Comment (IanRamosC) [inline client/components/marketing-survey/cancel-auto-renewal-form/index.jsx:21]: Nice catch. Thanks for improving that.

**Review by IanRamosC — APPROVED** [review]
The functionality tests well and code looks good. I tested the cancellation flow for Akismet products:
- purchased a month ago
- within refund window
As well as Jetpack products:
- within refund window
- disconnected Jetpack sites
- unattached licenses

Everything worked as expected. I just left a comment about a unit test that is breaking because of a change in this PR.

File: client/me/purchases/manage-purchase/purchase-meta-expiration.tsx [hunk 2]
```diff
@@ -66,13 +68,13 @@ function PurchaseMetaExpiration( {
...
+		const shouldRenderToggle = ( isCancellableSitelessPurchase || site ) && isProductOwner;
 
 		const autoRenewToggle = shouldRenderToggle ? (
 			<AutoRenewToggle
-				planName={ site.plan?.product_name_short }
-				siteDomain={ site.domain }
-				siteSlug={ site.slug }
+				planName={ isCancellableSitelessPurchase ? '' : site.plan?.product_name_short }
+				siteDomain={ isCancellableSitelessPurchase ? '' : site.domain }
+				siteSlug={ isCancellableSitelessPurchase ? '' : site.slug }
```
Comment (IanRamosC) [inline client/me/purchases/manage-purchase/purchase-meta-expiration.tsx]: It seems like this change is breaking some unit tests.

Comment (jboland88) [reply client/me/purchases/manage-purchase/purchase-meta-expiration.tsx]: Ah - thanks, getting that fixed 👍

**Review by CodeyGuyDylan — COMMENTED** [review]
Would it be a good idea to have a custom cancellation page here instead of the wordpress one? (As a future task)
image

File: client/me/purchases/manage-purchase/auto-renew-toggle/auto-renew-disabling-dialog/index.jsx [hunk 3]
```diff
@@ -132,6 +136,21 @@ class AutoRenewDisablingDialog extends Component {
...
+					'By canceling auto-renewal, your %(productName)s subscription will expire on %(expiryDate)s. ' +
+						"When it does, you'll lose access to key features you may be using on your site. " +
+						'To avoid that, turn auto-renewal back on or manually renew your subscription before the expiration date.',
+					{
+						args: {
+							productName: purchase.productName,
+							expiryDate,
+						},
+						comment:
+							'%(productName)s is the name of a WordPress.com product. ' +
```
Comment (CodeyGuyDylan) [inline client/me/purchases/manage-purchase/auto-renew-toggle/auto-renew-disabling-dialog/index.jsx]: In this case, wouldn't it always be an Akismet product?

**Review by CodeyGuyDylan — APPROVED** [review]
Other than a translator comment discrepancy, all looks good!

File: client/me/purchases/manage-purchase/auto-renew-toggle/auto-renew-disabling-dialog/index.jsx (hunk 3 above)
Comment (jboland88) [reply client/me/purchases/manage-purchase/auto-renew-toggle/auto-renew-disabling-dialog/index.jsx]: Ah, yep. Currently that is the case.

**Comment by jboland88** [comment]
> Would it be a good idea to have a custom cancellation page here instead of the wordpress one? (As a future task)

@CodeyGuyDylan - yes! I'd love to see a custom page/ flow here for Akismet in the future.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7892981

Thank you @jboland88 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7892981).

**Pull request merged**